js/ui/selection/selection.strategy.js

```diff
@@ -137,19 +137,20 @@ export default Class.inherit({
     _getFullSelectAllState: function() {
         const items = this.options.plainItems();
         const dataFilter = this.options.filter();
-        let selectedItems = this.options.selectedItems;
+        let selectedItems = this.options.ignoreDisabledItems ? this.options.selectedItems : this.options.selectedItems.filter(item => !item?.disabled);
 
         if(dataFilter) {
             selectedItems = dataQuery(selectedItems).filter(dataFilter).toArray();
         }
 
         const selectedItemsLength = selectedItems.length;
+        const disabledItemsLength = items.length - this.getSelectableItems(items).length;
```

Comment:
shouldn't it be disabled**Selected**ItemsLength?

Reply:
nope, we expect `true` when all non disabled items are selected
all non disabled items = all items - all disabled items